com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/scrbound.compute

```diff
@@ -1,538 +1,740 @@
 
-#include "Packages/com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightingConvexHullUtils.hlsl"
+    M._14_24_34 = d; // Last column
 
+    return M;
+}
 
-[numthreads(NR_THREADS, 1, 1)]
-void SCRAABBGEN(uint threadID : SV_GroupIndex, uint3 u3GroupID : SV_GroupID)
+// Scale followed by rotation (scaled axes).
+float3x3 ScaledRotation3x3(float3 xAxis, float3 yAxis, float3 zAxis)
 {
-    uint groupID = u3GroupID.x;
-    uint eyeIndex = u3GroupID.y; // currently, can only be 0 or 1
+    float3x3 R = float3x3(xAxis, yAxis, zAxis);
+    float3x3 C = transpose(R); // Row to column
 
-    // The g_ is preserved in order to make cross-pipeline (FPTL) updates easier
-    float4x4 g_mInvProjection = g_mInvProjectionArr[eyeIndex];
-    float4x4 g_mProjection = g_mProjectionArr[eyeIndex];
+    return C;
+}
 
-    //uint vindex = groupID * NR_THREADS + threadID;
-    unsigned int g = groupID;
-    unsigned int t = threadID;
+float3x3 Invert3x3(float3x3 R)
+{
+    float3x3 C   = transpose(R); // Row to column
+    float    det = dot(C[0], cross(C[1], C[2]));
+    float3x3 adj = float3x3(cross(C[1], C[2]),
+                            cross(C[2], C[0]),
+                            cross(C[0], C[1]));
+    return rcp(det) * adj;
+}
 
-    const int subLigt = (int) (t/8);
-    const int lgtIndex = subLigt+(int) g*8;
-    const int sideIndex = (int) (t%8);
+float4x4 Homogenize3x3(float3x3 R)
+{
+    float4x4 M = float4x4(float4(R[0], 0),
+                          float4(R[1], 0),
+                          float4(R[2], 0),
+                          float4(0,0,0,1));
+    return M;
+}
 
-    const int eyeAdjustedLgtIndex = GenerateLightCullDataIndex(lgtIndex, g_iNrVisibLights, eyeIndex);
-    SFiniteLightBound lgtDat = g_data[eyeAdjustedLgtIndex];
+float4x4 PerspectiveProjection4x4(float a, float g, float n, float f)
+{
+    float b = (f + n) * rcp(f - n);    // Z in [-1, 1]
+    float c = -2 * f * n * rcp(f - n); // No Z-reversal
 
-    const float3 boxX = lgtDat.boxAxisX.xyz;
-    const float3 boxY = lgtDat.boxAxisY.xyz;
-    const float3 boxZ = -lgtDat.boxAxisZ.xyz;           // flip axis (so it points away from the light direction for a spot-light)
-    const float3 center = lgtDat.center.xyz;
-    const float radius = lgtDat.radius;
-    const float2 scaleXY = lgtDat.scaleXY;
+    return float4x4(g/a, 0, 0, 0,
+                      0, g, 0, 0,
+                      0, 0, b, c,
+                      0, 0, 1, 0);
+}
 
-    {
-        if(sideIndex<6 && lgtIndex<(int) g_iNrVisibLights)      // mask 2 out of 8 threads
-        {
-            float3 q0, q1, q2, q3;
-            GetHullQuad(q0, q1, q2, q3, boxX, boxY, boxZ, center, scaleXY, sideIndex);
+/* ------------------------------ Implementation ---------------------------- */
 
+// Improve the quality of generated code at the expense of readability.
+// Remove when the shader compiler is clever enough to perform this optimization for us.
+#define DUMB_COMPILER
```

Comment:
always define? Even for PS4?

Reply:
maybe the name isn't good, we should say:
#define MANUAL_CODE_OPTIMIZATION ?

Reply:
I only tested on the PS4. Don't know about the XBox or Nvidia, I don't have the right hardware.

Reply:
ok, still we should avoid naming thing "dumb compiler" in public code :)